### PR TITLE
add container compile and insmod

### PR DESCRIPTION
**Pull request by lut777**
Now a bash script is available for quickly kernel module injection.

### CODE REVIEW

File: dist/images/install.sh [hunk 1]
```diff
@@ -2873,6 +2894,141 @@ dbtool(){
...
 }
 
+tuning(){
+  action="$1"; shift
+  sys="$1"; shift
+  case $action in
+    install-fastpath)
+      case $sys in
+        centos7)
+          docker run -it --privileged -v /lib/modules:/lib/modules -v /usr/src:/usr/src -v /tmp/:/tmp/ kubeovn/centos7-compile:"$OVN_VERSION" bash -c "./module.sh  centos install"
```
[the remark this answers is not in the document]
Comment (oilbeater) [reply dist/images/install.sh:2904]: why need privileged

File: Makefile [hunk 2]
```diff
@@ -51,6 +51,8 @@ base-arm64:
 release: lint build-go
 	docker buildx build --platform linux/amd64 --build-arg ARCH=amd64 -t $(REGISTRY)/kube-ovn:$(RELEASE_TAG) -o type=docker -f dist/images/Dockerfile dist/images/
 	docker buildx build --platform linux/amd64 --build-arg ARCH=amd64 -t $(REGISTRY)/vpc-nat-gateway:$(RELEASE_TAG) -o type=docker -f dist/images/vpcnatgateway/Dockerfile dist/images/vpcnatgateway
+	docker buildx build --platform linux/amd64 --build-arg ARCH=amd64 -t $(REGISTRY)/centos7-compile:$(RELEASE_TAG) -o type=docker -f dist/images/compile/centos7/Dockerfile fastpath/
+	docker buildx build --platform linux/amd64 --build-arg ARCH=amd64 -t $(REGISTRY)/centos8-compile:$(RELEASE_TAG) -o type=docker -f dist/images/compile/centos8/Dockerfile fastpath/
```
Comment (oilbeater) [inline Makefile:55]: need upload the image in github action

File: dist/images/kubectl-ko [hunk 3]
```diff
@@ -460,6 +468,141 @@ dbtool(){
...
+        *)
+          echo "unknown system $sys"
+      esac
+      ;;
+    remove-stt)
+      case $sys in
+        centos)
+          for i in $(kubectl -n kube-system get pods | grep ovn-cni | awk '{print $1}');
+          do
+            kubectl -n kube-system exec "$i" -- rm -f /tmp/*.rpm
```
Comment (oilbeater) [inline dist/images/kubectl-ko:594]: we should delete file carefully, use might also store other rpm in `/tmp`

File: pkg/daemon/controller.go [hunk 4]
```diff
@@ -1270,6 +1273,31 @@ func (c *Controller) deleteSubnetQos(subnet *kubeovnv1.Subnet) error {
 	return nil
 }
 
+func (c *Controller) operatemod() {
```
Comment (oilbeater) [inline pkg/daemon/controller.go:1276]: should use camelcase in all function and variable name

**Review by oilbeater — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (lut777) [reply Makefile:55]: make tar and tag&push action is added.

File: dist/images/compile/centos7/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,6 @@
+FROM centos:7
+
+COPY /*  /fastpath/
```
[the remark this answers is not in the document]
Comment (lut777) [reply dist/images/compile/centos7/Dockerfile:6]: Now all install & erase action are in one RUN action, and action COPY are moved to the end.

File: dist/images/compile/centos7/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+RUN yum install -y gcc elfutils-libelf-devel make perl python3 autoconf automake libtool rpm-build openssl-devel git
```
[the remark this answers is not in the document]
Comment (lut777) [reply dist/images/compile/centos7/Dockerfile]: Its a compile container and these packages are essential.

File: dist/images/compile/centos8/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,9 @@
+FROM centos:8
```
[the remark this answers is not in the document]
Comment (lut777) [reply dist/images/compile/centos8/Dockerfile:1]: compile container

File: dist/images/install.sh [hunk 8]
```diff
@@ -2411,6 +2424,7 @@ set -euo pipefail
 KUBE_OVN_NS=kube-system
 OVN_NB_POD=
 OVN_SB_POD=
+OVN_VERSION=
```
[the remark this answers is not in the document]
Comment (lut777) [reply dist/images/install.sh]: renamed.

File: dist/images/install.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (lut777) [reply dist/images/install.sh]: A variable REGISTRY has been introduced.
This container need to install package header-devel in the suitable distribution.

File: dist/images/kubectl-ko (hunk 3 above)
Comment (lut777) [reply dist/images/kubectl-ko]: Now we delete files like "openvswitch-kmod*.rpm"

File: pkg/daemon/controller.go (hunk 4 above)
Comment (lut777) [reply pkg/daemon/controller.go]: all variable and function names are renamed...

File: pkg/daemon/controller.go [hunk 9]
```diff
@@ -1315,3 +1345,92 @@ func recompute() {
...
+
+func removeAllMods(dir string) error {
+	kos, err := readKos(dir)
+	if err != nil {
+		return fmt.Errorf("access kos in %s failed: %v", dir, err)
+	}
+	for _, ko := range *kos {
+		err := removeKo(ko)
+		if err != nil {
+			klog.Errorf("remove module %s failed: %v", ko, err)
```
Comment (oilbeater) [inline pkg/daemon/controller.go:1378]: should return error here

File: .github/workflows/build-x86-image.yaml [hunk 10]
```diff
@@ -1151,8 +1163,16 @@ jobs:
           docker tag kubeovn/kube-ovn:$TAG kubeovn/kube-ovn:$TAG-x86
           docker tag kubeovn/vpc-nat-gateway:$TAG kubeovn/vpc-nat-gateway-dev:$COMMIT-x86
           docker tag kubeovn/vpc-nat-gateway:$TAG kubeovn/vpc-nat-gateway:$TAG-x86
+          docker tag kubeovn/centos7-compile:$TAG kubeovn/centos7-compile-dev:$TAG-x86
```
Comment (oilbeater) [inline .github/workflows/build-x86-image.yaml:1166]: should download and load the images first

File: dist/images/compile/centos7/Dockerfile [hunk 11]
```diff
@@ -0,0 +1,7 @@
+FROM centos:7
+
+RUN yum install -y gcc elfutils-libelf-devel make perl python3 autoconf automake libtool rpm-build openssl-devel git \
+    && git clone -b branch-2.16 --depth=1 https://github.com/openvswitch/ovs.git /ovs/ \
+    && yum erase -y git
```
Comment (oilbeater) [inline dist/images/compile/centos7/Dockerfile:5]: `yum clean all` is needed

File: dist/images/install.sh [hunk 12]
```diff
@@ -2771,6 +2787,12 @@ getOvnCentralPod(){
       exit 1
     fi
     OVN_SB_POD=$SB_POD
+    VERSION=$(kubectl  -n kube-system get pods -l ovn-sb-leader=true -o yaml | grep  "image: kubeovn/kube-ovn:" | head -n 1 | awk -F ':' '{print $3}')
```
Comment (oilbeater) [inline dist/images/install.sh:2790]: `$REGISTRY`

**Review by oilbeater — CHANGES_REQUESTED** [review]

File: dist/images/compile/centos7/Dockerfile (hunk 11 above)
Comment (lut777) [reply dist/images/compile/centos7/Dockerfile:5]: added

File: .github/workflows/build-x86-image.yaml (hunk 10 above)
Comment (lut777) [reply .github/workflows/build-x86-image.yaml:1166]: done.

File: pkg/daemon/controller.go (hunk 9 above)
Comment (lut777) [reply pkg/daemon/controller.go]: right...fixed.

**Pull request merged**